src/connector/src/parser/common.rs

```diff
@@ -42,16 +51,43 @@ fn do_parse_simd_json_value(dtype: &DataType, v: &BorrowedValue<'_>) -> Result<S
                 .map_err(|e| anyhow!("expect i32: {}", e))?,
         ),
         DataType::Int64 => ensure_int!(v, i64).into(),
-        DataType::Float32 => ScalarImpl::Float32((simd_json_ensure_float!(v, f32) as f32).into()),
-        DataType::Float64 => ScalarImpl::Float64((simd_json_ensure_float!(v, f64)).into()),
+        // when f32 overflows, the value is converted to `inf` which is inappropriate
+        DataType::Float32 => {
+            let scalar_val = ScalarImpl::Float32((simd_json_ensure_float!(v, f32) as f32).into());
+            if let ScalarImpl::Float32(f) = scalar_val {
+                if f.is_infinite() {
+                    anyhow::bail!("{v} is out of range for type f32");
+                }
```

Comment:
FYI this is okay only in the context of json, where the original value cannot be infinity.